### PR TITLE
Infixowl coverage ii

### PR DESCRIPTION
**Pull request by gjhiggins**
# Summary of changes
Final stage of update to infixowl, removing unused code, tidying, adding some docucomments, minor improvements in robustness, updating previously-failing tests, adding more tests. Expanded documentation to follow.

# Checklist

- [x] Checked that there aren't other open pull requests for the same change.
- [x] Added tests for any changes that have a runtime impact.
- [x] Checked that all tests and type checking passes.
- For changes that have a potential impact on users of this project:
  - [x] Updated relevant documentation to avoid inaccuracies.
  - [x] Considered adding additional documentation.
  - [x] Considered adding an example in `./examples` for new features.
  - [x] Considered updating our changelog (`CHANGELOG.md`).
- [x] Considered granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork),
  so maintainers can fix minor issues and keep your PR up to date.

### CODE REVIEW

File: rdflib/extras/infixowl.py [hunk 1]
```diff
@@ -131,63 +130,56 @@
...
 class Infix:
     def __init__(self, function):
         self.function = function
 
-    def __ror__(self, other):
-        return Infix(lambda x, self=self, other=other: self.function(other, x))
-
-    def __or__(self, other):
-        return self.function(other)
-
```
Comment (aucampia) [inline rdflib/extras/infixowl.py:190]: Can't we rather mark these as deprecated?

Comment (gjhiggins) [reply rdflib/extras/infixowl.py:190]: It's a bit late for that :smile: https://github.com/RDFLib/rdflib/blob/81b36e2968e18764d30c48295e3087f1836ecbe8/test/test_extras/test_infixowl/test_cover.py#L64

File: rdflib/extras/infixowl.py [hunk 2]
```diff
@@ -841,7 +894,7 @@ def deepClearIfBNode(_class):  # noqa: N802
         )
 
 
-class MalformedClass(Exception):
+class MalformedClassError(Exception):
```
Comment (aucampia) [inline rdflib/extras/infixowl.py:897]: This should be inhering from ValueError
```suggestion
class MalformedClassError(ValueError):
```

Comment (aucampia) [reply rdflib/extras/infixowl.py:897]: Also this makes it a breaking change.

Comment (gjhiggins) [reply rdflib/extras/infixowl.py:897]: The change was prompted by flake8 recommendation, maybe it's not worth making the change if it's a breaking one. Might as well revert it and re-do later for 0.7, what d'you reckon?

Comment (aucampia) [reply rdflib/extras/infixowl.py:897]: I will add the old class back in the type hierarchy and mark it as deprecated.

**Comment by aucampia** [comment]
@gjhiggins looks good, but I'm going to add release notes also.

File: rdflib/extras/infixowl.py (hunk 2 above)
Comment (aucampia) [reply rdflib/extras/infixowl.py]: Done now.

**Review by aucampia — APPROVED** [review]
Changes look good to me, please have a look at the changes I included for readme and the exception to see if you are fine with it.

**Comment by gjhiggins** [comment]
> Changes look good to me, please have a look at the changes I included for readme and the exception to see if you are fine with it.

Yep, fine by me. Extended documentation to come (honest, guv)

**Pull request merged**